langchain/src/experimental/prompts/handlebars.ts

```diff
@@ -0,0 +1,107 @@
+  const parsed: ParsedFStringNode[] = [];
+  const nodes: { type: string }[] = [...Handlebars.parse(template).body];
+  while (nodes.length) {
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    const node = nodes.pop()!;
+    if (node.type === "ContentStatement") {
+      // @ts-expect-error - handlebars' hbs.AST.ContentStatement isn't exported
+      const text = node.value;
+      parsed.push({ type: "literal", text });
+    } else if (node.type === "MustacheStatement") {
+      // @ts-expect-error - handlebars' hbs.AST.MustacheStatement isn't exported
+      const name: string = node.path.parts[0];
+      // @ts-expect-error - handlebars' hbs.AST.MustacheStatement isn't exported
+      const { original } = node.path as { original: string };
+      if (
+        !!name &&
+        !original.startsWith("this.") &&
+        !original.startsWith("@")
+      ) {
+        parsed.push({ type: "variable", name });
+      }
+    } else if (node.type === "PathExpression") {
+      // @ts-expect-error - handlebars' hbs.AST.PathExpression isn't exported
+      const name: string = node.parts[0];
+      // @ts-expect-error - handlebars' hbs.AST.PathExpression isn't exported
+      const { original } = node;
+      if (
+        !!name &&
+        !original.startsWith("this.") &&
+        !original.startsWith("@")
+      ) {
+        parsed.push({ type: "variable", name });
+      }
+    } else if (node.type === "BlockStatement") {
+      // @ts-expect-error - handlebars' hbs.AST.BlockStatement isn't exported
+      nodes.push([...node.params, ...node.program.body]);
+    }
+  }
+
+  return parsed;
+};
+
+export const interpolateHandlebars = (
+  template: string,
+  values: InputValues
+) => {
+  const compiled = Handlebars.compile(template, { noEscape: true });
+  return compiled(values);
+};
+
+export type HandlebarsPromptTemplateInput<RunInput extends InputValues> =
+  CustomFormatPromptTemplateInput<RunInput>;
+
+export class HandlebarsPromptTemplate<
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  RunInput extends InputValues = any
+> extends CustomFormatPromptTemplate<RunInput> {
+  static lc_name() {
+    return "CustomPromptTemplate";
+  }
+
+  lc_serializable = false;
+
+  constructor(input: CustomFormatPromptTemplateInput<RunInput>) {
+    super(input);
+    Object.assign(this, input);
+
+    if (this.validateTemplate && this.templateValidator !== undefined) {
+      let totalInputVariables: string[] = this.inputVariables;
+      if (this.partialVariables) {
+        totalInputVariables = totalInputVariables.concat(
+          Object.keys(this.partialVariables)
+        );
+      }
+      this.templateValidator(this.template, totalInputVariables);
+    }
+  }
+
+  /**
+   * Load prompt template from a template
+   */
+  static fromTemplate<
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    RunInput extends InputValues = any[]
+  >(
+    template: string,
+    params: Omit<
```

Comment:
Should omit `renderer` too, and could be defaulted to an empty object so that it doesn't need to be provided